referenceframe/frame.go

```diff
@@ -50,6 +50,7 @@ func RestrictedRandomFrameInputs(m Frame, rSeed *rand.Rand, lim float64) []Input
 		}
 
 		span := u - l
+		l = nodePose[i].Value - span/2
 		pos = append(pos, Input{lim*span*rSeed.Float64() + l + (span * (1 - lim) / 2)})
```

Comment:
This could now generate an out-of-bounds position. This function needs to guarantee returning in-bounds positions.

The simple solution is to cap the minimum/maximum values at the limits
```
if pos < l {
	pos = l
}
if pos > u {
	pos = u
}
```
The issue with this is that it weights the random position towards outputting a value at the frame limits.

The more complicated, and more correct, way to do this is to determine the valid range of values surrounding your input position, cap those ranges at l and u, and then use the random value to sample within that range.

Reply:
I'm not sure if I fully understand your explanation for the correct way to do this. I implemented what I understood but the way it is done now will mean that the node is no longer centered in the frame that is being sampled from, which could result in values being picked far away if the node is very close to the edge.